### PR TITLE
Refactor WebController

### PR DESCRIPTION
**Pull request by RickWieman**
Move pretty String builder of randomization to the RandomColorState.
Add toplevel commands without id (e.g. /random instead of just /random/{id})

### CODE REVIEW

**Comment by praseodym** [comment]
Looking good! I only have some doubts about the `PHBridge` return object in the facade. The facade is now still hard-coupled to the Philips Hue API, so we cannot just provide a dummy implementation. It is probably hard to refactor this (we'd need many more facade methods), but it might give us a nicer overall architecture.

**Comment by RickWieman** [comment]
I replaced/removed it in cdc241d :smiley:

File: src/main/java/ch/wisv/chue/hue/CHueFacade.java [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Component;
+
+import javax.annotation.PostConstruct;
+import java.util.List;
+import java.util.stream.Collectors;
+
+@Component
+public class CHueFacade implements HueFacade {
```
Comment (praseodym) [inline src/main/java/ch/wisv/chue/hue/CHueFacade.java]: This should be named `PhilipsHueFacadeImpl` or something more descriptive

**Comment by praseodym** [comment]
Looking good! Haven't had time to test it yet though.

**Comment by RickWieman** [comment]
Tested at CH, seems to work. :shipit:?

File: src/main/java/ch/wisv/chue/hue/HueLightState.java [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+import java.util.Optional;
+import java.util.OptionalInt;
+
+public class HueLightState {
+    private Optional<AlertMode> alertMode = Optional.empty();
+    private Optional<EffectMode> effectMode = Optional.empty();
+    private OptionalInt transitionTime = OptionalInt.empty();
+    private Optional<Color> color = Optional.empty();
+
+    protected static PHLightState asPHLightState(HueLightState hueLightState) {
```
Comment (praseodym) [inline src/main/java/ch/wisv/chue/hue/HueLightState.java]: This class shouldn't depend on `com.philips.lighting.*`, this method can be moved to `PhilipsHueFacade`.

File: src/main/java/ch/wisv/chue/hue/HueLightState.java [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+    }
+
+    public enum EffectMode {
+        NONE, COLORLOOP
+    }
+
+    @Override
+    public String toString() {
+        return "HueLightState{" +
+                "color=" + color +
```
Comment (praseodym) [inline src/main/java/ch/wisv/chue/hue/HueLightState.java:57]: This doesn't print anything useful

**Pull request merged**